### PR TITLE
Use Stop Code as Stop ID where Applicable

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR uses stop ID instead of stop code when it makes sense to. This makes the stop ID more useful for users.

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]
LGTM

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
The fallback to stop id doesn't seem to work in the default stops-overlay rendering.

File: packages/stops-overlay/src/default-stop-marker.js [hunk 1]
```diff
@@ -35,8 +35,8 @@ export default class StopMarker extends Component {
 
   render() {
     const { languageConfig, leafletPath, radius, stop } = this.props;
-    const { geometries, id, name, lat, lon } = stop;
-    const idArr = id.split(":");
+    const { code, id, geometries, lat, lon, name } = stop;
+    const userFacingId = code || id.split(":")[1];
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/stops-overlay/src/default-stop-marker.js]: This bug is also present in master, seems to be a case of the mock data not reflecting actual OTP responses. Either way, have fixed it here in 3e6aebba63fc90f9bbe3912bb8118507c2e14a6b

**Comment by miles-grant-ibigroup** [comment]
Thanks for the thorough review! All addressed

File: packages/stops-overlay/src/default-stop-marker.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/stops-overlay/src/default-stop-marker.js]: Nice!

**Pull request merged**